lib/matplotlib/colorbar.py

```diff
@@ -823,9 +818,8 @@ def _add_solids(self, X, Y, C):
 
     def _add_solids_pcolormesh(self, X, Y, C):
         _log.debug('Setting pcolormesh')
-        args = (X, Y, C) if self.orientation == 'vertical' else (Y.T, X.T, C.T)
```

Comment:
You don't need to do the transpose here any more?

Reply:
yes, because that just transposes how the array is represented internally in the pcolormesh, but doesn't change the mapping of what color gets mapped to what x/y box.